### ISSUE TITLE
Current State and Roadmap

### ISSUE THREAD

**Issue by ravage84** [cakephp/queue#25]
The current state of this plugin isn't mentioned anywhere. Also, there is no roadmap, as it seems.

For example, there hasn't been a single release, yet. Not even an alpha release.

**Comment by josegonzalez** [comment]
Good call. I think @markstory had some ideas/things he wanted to implement before an initial release :)

**Comment by markstory** [comment]
I would like to shore up some of the coverage gaps, and ensure we have at least a few end to end integration tests with one of the simple enqueue adapters before doing any releases.

As for a roadmap I can try to put something together later this week.

**Comment by markstory** [comment]
I had attempted to put the following in to the milestone description but GitHub butchers the formatting.

Initial release for the Queue Plugin.

## Goals

* Provide job consumer and Job classes to enable background job dispatching.
* Provide bake templates for job classes.

### Incomplete

* Decide final package name.
* Get to ~90% test coverage.
* Include integration tests with end to end job execution.
* Publish documentation.

**Comment by othercorey** [comment]
I request adding "Move under Cake/ namespace".

**Comment by josegonzalez** [comment]
I think I setup the plugin in the same way we have Migrations and Bake, neither of which have the `Cake/` namespace. Feel free to change it up though if we've decided that official plugins should have that, core or not :)

**Comment by othercorey** [comment]
I'm not going to make a big issue out of this, but my point-of-view is that we shouldn't be pushing things into the top-level namespace just because. I'm not a fan of the argument "well you shouldn't be using two queue systems in the first place.". Unless we're implementing de facto queue in php, our library names are too common.

**Comment by ADmad** [comment]
I added `Add Cake/ as top level namespace` to markstory's comment above. Our TwigView plugin also has the same.

**Comment by josegonzalez** [comment]
Sounds good :)

**Comment by markstory** [comment]
The namespace changes are done now :smile:

**Comment by T0mats0** [comment]
Looks like on the list is 90% coverage. We're almost there. QueueExtension needs a bit more. We also need end to end?
Without an integration test, I can't think of a way to to test QueueExtension without mocks. So what's left there?

We also talked about reducing static methods. I can imagine a lot here.

Documentation is left too. No idea what/how to write that up.

**Comment by markstory** [comment]
@T0mats0 I think an end to end test for a successful job dispatch and execution would be great to have. 

The documentation has been deployed and auto updates on each merge. It hasn't been linked into the make book table of contents as this plugin is still early on, but the docs are available at https://book.cakephp.org/queue

**Comment by kancers** [comment]
I'm really looking forward to the release. Is it almost there?
Thank you for all your wonderful OSS development.

**Comment by markstory** [comment]
I think we can do an initial release soon. The first batch of features are in place, and tests are much better than they were before.

**Comment by dereuromark** [comment]
Once this is done and docs are added, maybe also add it to https://github.com/FriendsOfCake/awesome-cakephp#queue for visibility?
